library/gen_x12_837.inc.php

```diff
@@ -282,585 +203,489 @@ function gen_x12_837($pid, $encounter, &$log, $encounter_claim = false)
+        $billingFacilityName = substr($claim->billingFacilityName(), 0, 60);
         $out .= "NM1" .       // Loop 2010AB Pay-To Provider
-        "*87" .
-        "*2" .
-        "*" . $billingFacilityName .
-        "*" .
-        "*" .
-        "*" .
-        "*";
+            "*" . "87" .
+            "*" . "2" .
+            "*" . $billingFacilityName .
+            "*" .
+            "*" .
+            "*" .
+            "*";
         if ($claim->billingFacilityNPI()) {
             $out .= "*XX*" . $claim->billingFacilityNPI();
-        } else {
-            $out .= "*24*" . $claim->billingFacilityETIN();
         }
-
         $out .= "~\n";
 
         ++$edicount;
         $out .= "N3" .
-        "*" . $claim->billingFacilityStreet() .
-        "~\n";
+            "*" . $claim->billingFacilityStreet() .
+            "~\n";
 
         ++$edicount;
         $out .= "N4" .
-        "*" . $claim->billingFacilityCity() .
-        "*" . $claim->billingFacilityState() .
-        "*" . stripZipCode($claim->billingFacilityZip()) .
-        "~\n";
+            "*" . $claim->billingFacilityCity() .
+            "*" . $claim->billingFacilityState() .
+            "*" . stripZipCode($claim->billingFacilityZip()) .
+            "~\n";
 
         if ($claim->billingFacilityNPI() && $claim->billingFacilityETIN()) {
             ++$edicount;
             $out .= "REF" .
-            "*EI" .
-            "*" . $claim->billingFacilityETIN() .
-            "~\n";
+                "*" . "EI" .
+                "*" . $claim->billingFacilityETIN() .
+                "~\n";
         }
     }
 
-  // Loop 2010AC Pay-To Plan Name omitted.  Includes:
-  // NM1*PE, N3, N4, REF*2U, REF*EI
+    // Loop 2010AC Pay-To Plan Name omitted.  Includes:
+    // NM1*PE, N3, N4, REF*2U, REF*EI
 
     $PatientHL = $claim->isSelfOfInsured() ? 0 : 1;
     $HLSubscriber = $HLcount++;
 
     ++$edicount;
     $out .= "HL" .        // Loop 2000B Subscriber HL Loop
-    "*$HLSubscriber" .
-    "*$HLBillingPayToProvider" .
-    "*22" .
-    "*$PatientHL" .
-    "~\n";
+        "*" . $HLSubscriber .
+        "*" . $HLBillingPayToProvider .
+        "*" . "22" .
+        "*" . $PatientHL .
+        "~\n";
 
     if (!$claim->payerSequence()) {
         $log .= "*** Error: Insurance information is missing!\n";
     }
 
     ++$edicount;
-    $out .= "SBR" .       // Subscriber Information
-    "*" . $claim->payerSequence() .
-    "*" . ($claim->isSelfOfInsured() ? '18' : '') .
-    "*" . $claim->groupNumber() .
-    "*" . (($CMS_5010 && $claim->groupNumber()) ? '' : $claim->groupName()) .
-    "*" . $claim->insuredTypeCode() . // applies for secondary medicare
-    "*" .
-    "*" .
-    "*" .
-    "*" . $claim->claimType() . // Zirmed replaces this
-    "~\n";
+    $out .= "SBR" .    // Subscriber Information
+        "*" . $claim->payerSequence() .
+        "*" . ($claim->isSelfOfInsured() ? '18' : '') .
+        "*" . $claim->groupNumber() .
+        "*" . $claim->groupName() .
```

Comment:
funny because was putting in group name only if there was no group number so think this is okay after dbl checking